File: backends/Auther/Utilities/ResponseFormatter.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Auther.Utilities
{
    public static class ResponseFormatter
    {
        public static IActionResult Success(string message, object? data = null)
        {
            return new JsonResult(new
            {
                status = "success",
                message,
                data
            })
            {
                StatusCode = StatusCodes.Status200OK
            };
        }

        public static IActionResult Created(string message, object? data = null)
        {
            return new JsonResult(new
            {
                status = "success",
                message,
                data
            })
            {
                StatusCode = StatusCodes.Status201Created
            };
        }

        public static IActionResult Error(int statusCode, string message)
        {
            return new JsonResult(new
            {
                status = "error",
                message
            })
            {
                StatusCode = statusCode
            };
        }
    }
}

File: backends/fastapi/app/utils/response_formatter.py
from fastapi.responses import JSONResponse

class ResponseFormatter:
    @staticmethod
    def success(message: str, data: dict = None):
        return JSONResponse(
            content={
                "status": "success",
                "message": message,
                "data": data,
            },
            status_code=200
        )

    @staticmethod
    def created(message: str, data: dict = None):
        return JSONResponse(
            content={
                "status": "success",
                "message": message,
                "data": data,
            },
            status_code=201
        )

    @staticmethod
    def error(message: str, status_code: int):
        return JSONResponse(
            content={
                "status": "error",
                "message": message,
            },
            status_code=status_code
        )
